cookbooks/calico/recipes/general_compute.rb

```diff
@@ -4,6 +4,18 @@
 # Find the BGP neighbors, which is everyone except ourselves.
 bgp_neighbors = search(:node, "role:compute").select { |n| n[:ipaddress] != node[:ipaddress] }
 
+# Overwrite chef's IPv6 address assignment because there's more than one to choose from.
+ruby_block "assign-bgp-ipv6-address" do
+    block do
+        addresses = node[:network][:interfaces][:eth0][:addresses]
+        global_ipv6 = addresses.select do |address|
+            address[:family] == 'inet6' && address[:scope] == 'Global'
+        end
+        node[:bgp_ip6address] = global_ipv6.keys.sort[0].to_s
+    end
+    action [:nothing]
```

Comment:
you probably only want to do this if the address has not already been determined.